### PR TITLE
Bug 1668186 - Allow skipping avoidable pushes

### PR DESCRIPTION
**Pull request by mozfreddyb**

### CODE REVIEW

File: libmozevent/mercurial.py [hunk 1]
```diff
@@ -305,6 +307,24 @@ async def run(self):
                     "Unsupported repository", repo=build.repo_phid, build=build
                 )
 
+    def is_commit_skippable(self, build):
+        def get_files_touched_in_diff(rawdiff):
+            patched = []
+            for parsed_diff in rs_parsepatch.get_diffs(rawdiff):
+                # filename is sometimes of format 'test.txt  Tue Feb 05 17:23:40 2019 +0100'
```
Comment (marco-c) [inline libmozevent/mercurial.py:314]: Could you point to the issue you file in rs_parsepatch?

File: libmozevent/mercurial.py [hunk 2]
```diff
@@ -305,6 +307,24 @@ async def run(self):
...
+                    patched.append(filename)
+            return patched
+
+        for rev in build.stack:
+            files = get_files_touched_in_diff(rev.patch)
+            for patched_file in files:
+                if patched_file in self.skippable_files:
+                    # early return because one file touched is enough for us to bail out
+                    return True
+        return False
```
Comment (marco-c) [inline libmozevent/mercurial.py:326]: Nicer Pythonic way:
```
return any(patched_file in self.skippable_files for rev in build.stack for patched_file in get_files_touched_in_diff(rev.patch))
```

**Review by marco-c — APPROVED** [review]
LGTM, could you rebase on top of current master?

File: libmozevent/mercurial.py (hunk 1 above)
Comment (mozfreddyb) [reply libmozevent/mercurial.py:314]: yup, will do.

File: libmozevent/mercurial.py (hunk 2 above)
Comment (mozfreddyb) [reply libmozevent/mercurial.py:326]: Not sure I would agree with the double-loop as being more readable, but happy to take that. Updated PR coming.

**Comment by mozfreddyb** [comment]
That should do it. Thanks a lot!

**Comment by mozfreddyb** [comment]
FWIW, this is related to https://bugzilla.mozilla.org/show_bug.cgi?id=1668186

**Pull request merged**